Fix tiltrotor motor tilt oscillations

Jaeyoung-Lim opened the pull request:
This PR fixes the tilt rotor motor tilts oscillating during flight and even during the vehicle is stationary. This was causing the joint position to be more or less random and was blocking a lot of functionalities to work that require joint position to be precise (e.g. controlling yaw based on tiltrotor tilt). Also, the flight shows instabilities due to the motors pointing in arbitrary directions during flight.

The oscillations were caused by the following reasons.

- Joint controllers becoming unstable very quickly due to large control errors with large gains.
- Inertia of the joints were too small
- Poor handling of joint limits in combination of actuator input scaling

To solve the problem the following is proposed
- Increase of motor inertia 
- Added friction to the tilt joints
- Clamp error of the joint controller to prevent pid controller becoming unstable with large errors

**Before PR**
![ezgif com-optimize(1)](https://user-images.githubusercontent.com/5248102/80366686-f628fd80-8889-11ea-86bc-a57029d71069.gif)


**After PR**
![ezgif com-optimize](https://user-images.githubusercontent.com/5248102/80366697-f923ee00-8889-11ea-9a11-a980158e3412.gif)

Jaeyoung-Lim committed:
Fix tiltrotor oscillations

Fix tiltrotor joint oscillations

sfuhrer reviewed (approved):
It's much better with these changes, especially using tilt support in hover and transitions. Awesome @Jaeyoung-Lim !

bkueng commented on src/gazebo_mavlink_interface.cpp:
```diff
@@ -1531,6 +1531,7 @@ void GazeboMavlinkInterface::handle_control(double _dt)
 #endif
 
         double err = current - target;
+        err = std::max(std::min(err, 0.2), -0.2);
```
0.2 looks like a magic number.

sfuhrer replied:
@Jaeyoung-Lim could you add a comment there why the 0.2 is necessary/how you got to it? Then we get this in right away I would say, as it clearly helps against the propeller shaking.

Jaeyoung-Lim replied:
@bkueng I added a parameter that you can set the maximum joint errors

bkueng commented on src/gazebo_mavlink_interface.cpp:
```diff
@@ -156,6 +156,7 @@ void GazeboMavlinkInterface::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf
   input_reference_.resize(n_out_max);
   joints_.resize(n_out_max);
   pids_.resize(n_out_max);
+  joint_max_errors_.resize(n_out_max);
```
Would be better to have a single `vector` for those 3, but given it's already done that way, no need to change here.

Jaeyoung-Lim replied:
Right, makes sense.

bkueng commented on src/gazebo_mavlink_interface.cpp:
```diff
@@ -1531,7 +1535,9 @@ void GazeboMavlinkInterface::handle_control(double _dt)
 
         double err = current - target;
-        err = std::max(std::min(err, 0.2), -0.2);
+        if(joint_max_errors_[i]!=0) {
```
```suggestion
        if(joint_max_errors_[i]!=0.) {
```

Jaeyoung-Lim replied:
Thanks!

Jaeyoung-Lim committed:
Add maximum joint error as paramters

This removes the magic number of max joint errors to be able to explicitly define the maximum error handled for the pid controller in the joints
